src/notfound/handler.js

```diff
@@ -29,18 +32,50 @@ function process404Response(data) {
       source: row.source,
     }));
 }
+
+/**
+ * Processes the audit by fetching site data,PSI data, code diff and content last modified date
+ * creating audit data, and sending a message to SQS.
+ *
+ * @async
+ * @param {Object} dataAccess - Object containing the functions supported by the data
+ * @param {Object} site - The site which to audit.
+ * @param {Object} auditContext - The audit context object containing information about the audit.
+ * @param {Object} result - The result object containing audit result.
+ * @throws {Error} - Throws an error if any step in the audit process fails.
+ */
+async function processAuditResult(
+  dataAccess,
+  site,
+  auditContext,
+  result,
+) {
+  const auditData = {
+    siteId: site.getId(),
+    auditType: AUDIT_TYPE,
+    auditedAt: new Date().toISOString(),
+    isLive: site.isLive(),
+    auditResult: { ...result, finalUrl: auditContext.finalUrl },
+  };
+
+  await dataAccess.addAudit(auditData);
+}
 export default async function audit404(message, context) {
-  const { type, url, auditContext } = message;
-  const { log, sqs } = context;
+  const { type, url: siteId, auditContext } = message;
+  const { log, sqs, dataAccess } = context;
   const {
     AUDIT_RESULTS_QUEUE_URL: queueUrl,
   } = context.env;
 
   try {
-    log.info(`Received audit req for domain: ${url}`);
+    log.info(`Received audit req for domain: ${siteId}`);
```

Comment:
Should it be `for site id`?

Reply:
also, we should ideally always log both the siteId and the baseURL - so we can more easily extract / filter on coralogix.